src/components/Tool/Appeal/InitialPage/AddAppealForm/AddAppealForm.tsx

```diff
@@ -389,217 +400,251 @@ const AddAppealForm: React.FC<AddAppealFormProps> = ({
-                </Box>
-              </Grid>
-              <Grid item xs={12} sm={2}>
-                <Box
-                  display="flex"
-                  flexDirection="column"
-                  justifyContent="start"
-                >
-                  <Field
-                    name="letterCost"
-                    type="number"
-                    variant="outlined"
-                    label={t('Letter Cost')}
-                    size="small"
-                    className={classes.input}
-                    error={errors.letterCost}
-                    helperText={errors.letterCost}
-                    as={TextField}
-                  />
-                </Box>
-              </Grid>
-              <Grid item xs={12} sm={1}>
-                <Box
-                  display="flex"
-                  alignItems="center"
-                  justifyContent="center"
-                  style={{
-                    height: '100%',
-                  }}
-                >
-                  <Icon path={mdiClose} size={1} />
-                </Box>
-              </Grid>
-              <Grid item xs={12} sm={2}>
-                <Box
-                  display="flex"
-                  flexDirection="column"
-                  justifyContent="start"
-                >
-                  <Field
-                    name="adminPercentage"
-                    type="number"
-                    variant="outlined"
-                    label={t('Admin %')}
-                    size="small"
-                    className={classes.input}
-                    error={errors.adminPercentage}
-                    helperText={errors.adminPercentage}
-                    as={TextField}
-                  />
-                </Box>
-              </Grid>
-              <Grid item xs={12} sm={1}>
-                <Box
-                  display="flex"
-                  alignItems="center"
-                  justifyContent="center"
-                  style={{
-                    height: '100%',
-                  }}
-                >
-                  <Icon path={mdiEqual} size={1} />
-                </Box>
-              </Grid>
-              <Grid item xs={12} sm={3}>
-                <Box
-                  display="flex"
-                  flexDirection="column"
-                  justifyContent="start"
-                >
-                  <TextField
-                    data-testid="goalInput"
-                    name="goal"
-                    type="number"
-                    label={t('Goal')}
-                    variant="outlined"
-                    size="small"
-                    className={classes.input}
-                    value={calculateGoal(
-                      initialGoal,
-                      letterCost,
-                      adminPercentage,
-                    ).toFixed(2)}
-                  />
-                </Box>
+      }): ReactElement => {
+        useEffect(() => {
```

Comment:
I guess this works without errors, but I don't love that we're using hooks not at the top-level of the component. Can we accept a little bit of duplication and set the goal field value in the `onChange` handlers for the other 3 inputs?

Reply:
I think, technically, it is at the top of the Formik component. But I can remove it and duplicate the code.